### ISSUE TITLE
`cdb_observatory._OBS_GetGeometryScores` is returning incorrect results for points and small polygons

### ISSUE THREAD

**Issue by antoniocarlon** [CartoDB/bigmetadata#243]
The `cdb_observatory._OBS_GetGeometryScores` is returning weird results when querying it using points or small polygons.

For example:

```
select cdb_observatory._OBS_GetGeometryScores(
'0103000020E61000000100000021000000DF38AA76DED55EC0E04B046EE937484041F0B546DFD55EC0CAB627ECDD374840070498C2E1D55EC0D95CE9D8D237484098C4DED1E5D55EC0D05F3DA1C83748407792984CEBD55EC0CD53A7A9BF3748407F07DDFCF1D55EC0AD955D4AB8374840AB60DFA0F9D55EC07AB0E5CBB23748409DC375ED01D65EC046204B64AF3748405582FC900AD65EC0BED20B35AE374840D2FB783613D65EC0A7CACE49AF374840EF49DD881BD65EC09089F197B23748406DAC4C3623D65EC004A5F0FEB7374840C79940F329D65EC02084A749BF374840E7906F7D2FD65EC094005B30C8374840B227599E33D65EC000DD7A5BD2374840234A5E2D36D65EC0E2FCFE66DD374840230C511137D65EC0734E3FE6E837484038A96C4136D65EC0079E2068F4374840311FACC533D65EC025456D7BFF3748408C027CB62FD65EC0A7F42EB3093848406580CA3B2AD65EC0D3CADEAA1238484094BB7E8B23D65EC0AB7D420A1A384840109466E71BD65EC08D8ED0881F3848401736AF9A13D65EC055217AF022384840065501F70AD65EC056FFBE1F2438484014875D5102D65EC0EE4CF70A2338484038AFD7FEF9D55EC0D940C6BC1F384840CFA85151F2D55EC0EC72B1551A384840BB6F5594EBD55EC0A7C9E00A1338484064C82D0AE6D55EC07B5813240A384840F1FF59E9E1D55EC06050DDF8FF37484087D8755ADFD55EC0EA2D4AEDF4374840DF38AA76DED55EC0E04B046EE9374840',
'{ca.statcan.geo.cd_,ca.statcan.geo.cma,ca.statcan.geo.csd,ca.statcan.geo.ct_,ca.statcan.geo.pr_}',
500
)
```

Returns:

```
'(0.871437099846586265576905188325662646610,1,ca.statcan.geo.geography_cma_42eb24c786,ca.statcan.geo.cma,,0.0000118849668205594,0.06666666666666666667,,)'
'(0.8676192883315463467393809644165773173348,1,ca.statcan.geo.geography_cd_e928e96afb,ca.statcan.geo.cd_,,0.00000354004208463742,0.06666666666666666667,,)'
'(0.8659203712971341531551226852315233743222580,1,ca.statcan.geo.geography_pr_ce37f70250,ca.statcan.geo.pr_,,0.00000000754508371298107,0.06666666666666666667,,)'
'(0.84613633623968853337791925243276539984,1,ca.statcan.geo.geography_csd_06e4ed8015,ca.statcan.geo.csd,,0.000458411780353777,0.05882352941176470588,,)'
'(0.7594808250608149153605722192516448345,1,ca.statcan.geo.geography_ct_3d994ee11d,ca.statcan.geo.ct_,,0.00308263518425256,0.04705882352941176471,,)'
```

According to this scores, the correct resolution will be CMA and it seems that CT_ would be much better.

Other cases to test:

```
SELECT cdb_observatory.OBS_getmeasure(ST_SetSRID(ST_Point(-123.37423325, 48.43577815),4326),'ca.statcan.cols_census.t001c001_t','point','ca.statcan.geo.ct_')
```

returns (because it forces CT_ to be used)

```
3123.15409580865
```

but 

```
SELECT cdb_observatory.OBS_getmeasure(ST_SetSRID(ST_Point(-123.37423325, 48.43577815),4326),'ca.statcan.cols_census.t001c001_t')
```

returns 

```
489.253484023405
```

because it relays on the `cdb_observatory._OBS_GetGeometryScores` that computes an incorrect score for each resolution

**Comment by javitonino** [comment]
Note that if you look at the `GetGeometryScores` output, the last column indicates the % of the raster pixel filled by polygons. In this case, it is returning different values for CSD, but that should have the same coverage as the others (0.06). It appears there is something wrong in raster generation.

**Comment by antoniocarlon** [comment]
Solved in #244

Querying `cdb_observatory._OBS_GetGeometryScores` using a small geometry.

```
select cdb_observatory._OBS_GetGeometryScores(
'0103000020E61000000100000021000000DF38AA76DED55EC0E04B046EE937484041F0B546DFD55EC0CAB627ECDD374840070498C2E1D55EC0D95CE9D8D237484098C4DED1E5D55EC0D05F3DA1C83748407792984CEBD55EC0CD53A7A9BF3748407F07DDFCF1D55EC0AD955D4AB8374840AB60DFA0F9D55EC07AB0E5CBB23748409DC375ED01D65EC046204B64AF3748405582FC900AD65EC0BED20B35AE374840D2FB783613D65EC0A7CACE49AF374840EF49DD881BD65EC09089F197B23748406DAC4C3623D65EC004A5F0FEB7374840C79940F329D65EC02084A749BF374840E7906F7D2FD65EC094005B30C8374840B227599E33D65EC000DD7A5BD2374840234A5E2D36D65EC0E2FCFE66DD374840230C511137D65EC0734E3FE6E837484038A96C4136D65EC0079E2068F4374840311FACC533D65EC025456D7BFF3748408C027CB62FD65EC0A7F42EB3093848406580CA3B2AD65EC0D3CADEAA1238484094BB7E8B23D65EC0AB7D420A1A384840109466E71BD65EC08D8ED0881F3848401736AF9A13D65EC055217AF022384840065501F70AD65EC056FFBE1F2438484014875D5102D65EC0EE4CF70A2338484038AFD7FEF9D55EC0D940C6BC1F384840CFA85151F2D55EC0EC72B1551A384840BB6F5594EBD55EC0A7C9E00A1338484064C82D0AE6D55EC07B5813240A384840F1FF59E9E1D55EC06050DDF8FF37484087D8755ADFD55EC0EA2D4AEDF4374840DF38AA76DED55EC0E04B046EE9374840',
'{ca.statcan.geo.cd_,ca.statcan.geo.cma,ca.statcan.geo.csd,ca.statcan.geo.ct_,ca.statcan.geo.pr_}',
500
) order by 1 desc
```

returns correct results as it gives more precedence to CT_ than any other resolution and the order of the resolutions are correct (CT>CSD>CMA>CD>PR):

```
'(1.1389030579245615833130993646908995558,1,ca.statcan.geo.geography_ct_3d994ee11d,ca.statcan.geo.ct_,,0.00306997479012511,0.07058823529411764706,,)'
'(1.01543212438943520665221798638688274364,1,ca.statcan.geo.geography_csd_06e4ed8015,ca.statcan.geo.csd,,0.000459015294904344,0.07058823529411764706,,)'
'(0.922754844235905664130673365050990584598,1,ca.statcan.geo.geography_cma_42eb24c786,ca.statcan.geo.cma,,0.0000120062182815153,0.07058823529411764706,,)'
'(0.9186708416856993844578994159888793034662,1,ca.statcan.geo.geography_cd_e928e96afb,ca.statcan.geo.cd_,,0.00000357020627087331,0.07058823529411764706,,)'
'(0.916856974649332449428709184871606521215714,1,ca.statcan.geo.geography_pr_ce37f70250,ca.statcan.geo.pr_,,0.0000000077595700059694,0.07058823529411764706,,)'
```

Also, querying `cdb_observatory.OBS_getmeasure` using a point and forcing a resolution returns the same value as querying `cdb_observatory.OBS_getmeasure` relying on `cdb_observatory._OBS_GetGeometryScores` to calculate the scores for each resolution (the following queries return the same result)

```
SELECT cdb_observatory.OBS_getmeasure(ST_SetSRID(ST_Point(-123.37423325, 48.43577815),4326),'ca.statcan.cols_nhs.t001c001_t','point','ca.statcan.geo.ct_')

SELECT cdb_observatory.OBS_getmeasure(ST_SetSRID(ST_Point(-123.37423325, 48.43577815),4326),'ca.statcan.cols_nhs.t001c001_t')
```
